ethereum/blockchain_lifecycle.go

```diff
@@ -0,0 +1,43 @@
+package ethereum
+
+import (
+	"context"
+	"testing"
+
+	"github.com/ethereum/go-ethereum/rpc"
+	"github.com/stretchr/testify/require"
+)
+
+type BlockchainLifecycle struct {
+	rpcClient       *rpc.Client
+	snapshotIdStack []string
+}
+
+func NewBlockchainLifecycle(rpcURL string) (*BlockchainLifecycle, error) {
```

Comment:
Sorry I forgot earlier. We should probably add some documentation to this file. I recognize this from the TypeScript version, but other people reading/contributing to the code might not get it right away.